### ISSUE TITLE
False positive: SC2016: `find . -exec dash -c ...` complains about single quoted expressions.

### ISSUE THREAD

**Issue by Dylan-Brotherston** [koalaman/shellcheck#2788]
#### For bugs
- Rule Id (if any, e.g. SC1000): CS2016 
- My shellcheck version (`shellcheck --version` or "online"): online
- [X] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### Here's a snippet or screenshot that shows the problem:

```sh
#! /usr/bin/env dash

find . -exec bash -c 'echo "$1"' shell {} \; # fine

find . -exec sh -c 'echo "$1"' shell {} \; # fine

find . -exec dash -c 'echo "$1"' shell {} \; # CS2016 warning

# shellcheck disable=SC2016
find . -exec dash -c 'echo "$1"' shell {} \; # warning disabled for this line
```

#### Here's what shellcheck currently says:
```
Line 7:
find . -exec dash -c 'echo "$1"' shell {} \; # CS2016 warning
                     ^-- SC2016 (info): Expressions don't expand in single quotes, use double quotes for that.
```

#### Here's what I wanted or expected to see:

No `CS2016` info message raised at line 7.

from the wiki page for CS2016
> ShellCheck tries to increase the signal-to-noise ratio of this warning by ignoring certain well known commands that frequently expect literal dollar signs, such as sh and perl. However, there's a long tail of less common commands and flags that also frequently expect $s, and it's not in ShellCheck's scope to try to keep track of them all.

but as `find . -exec bash -c` and `find . -exec sh -c` are already exceptions, `find . -exec dash -c` should be as well.
